core/src/main/scala/cromwell/core/actor/BatchingDbWriter.scala

```diff
@@ -0,0 +1,51 @@
+package cromwell.core.actor
+
+import cats.data.NonEmptyVector
+import org.slf4j.LoggerFactory
+
+import scala.util.{Failure, Success, Try}
+
+
+/** A collection of state, data, and message types to support batched database writes. */
+object BatchingDbWriter {
+  val logger = LoggerFactory.getLogger("BatchingDbWriteActor")
+
+  /** Data for batched database writes. */
+  sealed trait BatchingDbWriterData {
+    def addData[D](datum: D): BatchingDbWriterData = addData(Vector(datum))
+    def addData[D](data: Iterable[D]): BatchingDbWriterData = {
+      Try(NonEmptyVector.fromVector(data.toVector)) match {
+        case Success(Some(v)) =>
+          val newEvents = this match {
+            case NoData => v
+            case HasData(e) => e.concatNev(v)
+          }
+          HasData(newEvents)
+        case Success(None) => this
+        case Failure(f) =>
+          val dataSample = data.mkString(", ") + (if (data.size > 3) ", ..." else "")
```

Comment:
Am I missing something? This doesn't look like it's truncating `data` for the `mkString`, but your `...` makes it look like it should be?